tests/cmds/test_cases.py

```diff
@@ -0,0 +1,279 @@
+    cli_state.sdk.cases.file_events.get_all.assert_called_once_with(1)
+    assert "Invalid case-number 1." in result.output
+
+
+def test_file_events_add_when_py42_raises_exception_returns_error_message(
+    runner, cli_state, error
+):
+    cli_state.sdk.cases.file_events.add.side_effect = Py42BadRequestError(error)
+    result = runner.invoke(
+        cli, ["cases", "file-events", "add", "1", "--event-id", "1"], obj=cli_state,
+    )
+    cli_state.sdk.cases.file_events.add.assert_called_once_with(1, "1")
+    assert "Invalid case-number or event-id." in result.output
+
+
+def test_file_events_remove_when_py42_raises_exception_returns_error_message(
+    runner, cli_state, error
+):
+    cli_state.sdk.cases.file_events.delete.side_effect = Py42NotFoundError(error)
+    result = runner.invoke(
+        cli, ["cases", "file-events", "remove", "1", "--event-id", "1"], obj=cli_state,
+    )
+    cli_state.sdk.cases.file_events.delete.assert_called_once_with(1, "1")
+    assert "Invalid case-number or event-id." in result.output
+
+
+def test_show_returns_expected_data(runner, cli_state, py42_response):
+    py42_response.data = json.loads(CASE_DETAILS)
+    cli_state.sdk.cases.get.return_value = py42_response
+    result = runner.invoke(cli, ["cases", "show", "1"], obj=cli_state,)
+    assert "test@test.test" in result.output
+
+
+def test_list_returns_expected_data(runner, cli_state, py42_response, mocker):
+    py42_response.data = json.loads(ALL_CASES)
+
+    def gen():
+        yield py42_response.data
+
+    cli_state.sdk.cases.get_all.return_value = gen()
+    result = runner.invoke(cli, ["cases", "list"], obj=cli_state,)
+    assert "test@test.test" in result.output
+
+
+def test_show_returns_expected_data_with_include_file_events_option(
+    runner, cli_state, py42_response
+):
+    py42_response.text = ALL_EVENTS
+    cli_state.sdk.cases.file_events.get_all.return_value = py42_response
+    result = runner.invoke(
+        cli, ["cases", "show", "1", "--include-file-events"], obj=cli_state,
+    )
+    assert (
+        "0_1d71796f-af5b-4231-9d8e-df6434da4663_984418168383179707_986472527798692818_971"
+        in result.output
+    )
+
+
+def test_events_list_returns_expected_data(runner, cli_state):
+    cli_state.sdk.cases.file_events.get_all.return_value = json.loads(ALL_EVENTS)
+    result = runner.invoke(cli, ["cases", "file-events", "list", "1"], obj=cli_state,)
+    assert (
+        "0_1d71796f-af5b-4231-9d8e-df6434da4663_984418168383179707_986472527798692818_971"
+        in result.output
+    )
+
+
+@pytest.mark.parametrize(
+    "command, error_msg",
+    [
+        ("{} create --description d".format(CASES_COMMAND), MISSING_NAME),
+        ("{} update --description d".format(CASES_COMMAND), MISSING_CASE_NUMBER),
+        ("{} show".format(CASES_COMMAND), MISSING_CASE_NUMBER),
+        ("{} export".format(CASES_COMMAND), MISSING_CASE_NUMBER),
+        # ("{} add ".format(CASES_FILE_EVENTS_COMMAND), MISSING_CASE_NUMBER),
```

Comment:
This test and the 2 others commented below, fails with error `Error: Invalid value for 'CASE_NUMBER':  is not a valid integer\n` when executed through `pytest` but on command line the return expected `MISSING_CASE_NUMBER` error message. 

Not sure whether to change the test expectation or keep the test!!

Reply:
The reason for this is because you are ending these commands with spaces.
When you call `split(" ")`, it has an extra space-arg at the end, which it thinks is the case numbers.  Obviously, empy space is not a number.

Do this: `"{} add".format(CASES_FILE_EVENTS_COMMAND), MISSING_CASE_NUMBER),`
instead of this `"{} add ".format(CASES_FILE_EVENTS_COMMAND), MISSING_CASE_NUMBER),`

Reply:
Eagle's eye. Thanks!